lib/phase/merge.js

```diff
@@ -79,6 +79,32 @@ function mergeSharedConfig(parsedDoc, pipelineTemplate) {
         newPipeline.shared.image = mergedImage;
     }
 
+    if (parsedDoc.cache) {
+        newPipeline.cache = {
+            ...newPipeline.cache,
+            ...parsedDoc.cache
+        };
+    }
+
+    newPipeline.subscribe = {
+        ...newPipeline.subscribe,
+        ...parsedDoc.subscribe
+    };
```

Comment:
Same here. Instead of high level replace (i.e. pipeline `subscribe.scmUrls` completely replacing pipeline template `subscribe.scmUrls` ), should we append the pipeline `subscribe.scmUrls` to pipeline template `subscribe.scmUrls`?

Reply:
this honestly it could go both ways, what I was thinking is that user may or maybe not want subscribe defined by pipeline template, if they want to retain that same scmurl while adding new ones, then they should add both to their yaml. otherwise how do you handle the case where user that just want to replace it